### PR TITLE
New Stationware Events

### PR DESCRIPTION
**Pull request by Just-a-Unity-Dev**
## About the PR

Adds six(?) new events to StationWare

**Media**

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: eclips_e
- add: Added a bunch of new events because I was evicted from the host mines

### CODE REVIEW

File: Content.Server/_StationWare/Challenges/Modifiers/Systems/RequireSlotOccupiedSystem.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                continue;
+
+            if (component.RequiredPrototype == null)
+            {
+                _stationWareChallenge.SetPlayerChallengeState(ent, uid, true, args.Component);
+            }
+            else
+            {
+                // assume we need to check prototype
+                if (meta_data.EntityPrototype.ID == component.RequiredPrototype)
```
Comment (EmoGarbage404) [inline Content.Server/_StationWare/Challenges/Modifiers/Systems/RequireSlotOccupiedSystem.cs:63]: don't compare entity IDs you donkey. Just use an EntityWhitelist and add a tag.

Comment (Just-a-Unity-Dev) [reply Content.Server/_StationWare/Challenges/Modifiers/Systems/RequireSlotOccupiedSystem.cs:63]: waa

File: Resources/Prototypes/Challenges/challenges.yml [hunk 2]
```diff
@@ -163,33 +163,42 @@
...
-    - id: ClothingHeadHatWizard
-      orGroup: hat
-      prob: 0.33
-    - id: ClothingHeadHatVioletwizard
-      orGroup: hat
-      prob: 0.33
-    - id: ClothingHeadHatRedwizard
-      orGroup: hat
-      prob: 0.33
+      - id: SmiteBook
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Challenges/challenges.yml:166]: why did you fuck up the indentation.

Comment (Just-a-Unity-Dev) [reply Resources/Prototypes/Challenges/challenges.yml:166]: cause rider

File: Content.Server/_StationWare/Challenges/Modifiers/Components/RequireSlotOccupiedComponent.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Content.Server._StationWare.Challenges.Modifiers.Components;
+
+/// <summary>
+/// This is used for events to check whether their slot has a prototype or is generally occupied
+/// </summary>
+[RegisterComponent]
+public sealed class RequireSlotOccupiedComponent : Component
+{
+    [DataField("slot")] public string Slot = "head";
```
Comment (EmoGarbage404) [inline Content.Server/_StationWare/Challenges/Modifiers/Components/RequireSlotOccupiedComponent.cs:11]: make both of these datafields required
[DataField("slot", required: true)]

File: Content.Server/_StationWare/Challenges/Modifiers/Systems/RequireSlotOccupiedSystem.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        SubscribeLocalEvent<RequireSlotOccupiedComponent, ChallengeStartEvent>(OnChallengeStart);
+        SubscribeLocalEvent<RequireSlotOccupiedComponent, BeforeChallengeEndEvent>(OnBeforeChallengeEnd);
+    }
+
+    private void OnChallengeStart(EntityUid uid, RequireSlotOccupiedComponent component,
+        ref ChallengeStartEvent args)
+    {
+        foreach (var player in args.Players)
+        {
+            var winning = _componentFactory.GetComponent<RequireSlotOccupiedTrackerComponent>();
```
Comment (EmoGarbage404) [inline Content.Server/_StationWare/Challenges/Modifiers/Systems/RequireSlotOccupiedSystem.cs:28]: just add the comp normally here.

File: Resources/Locale/en-US/_StationWare/ware-events/challenges.ftl [hunk 5]
```diff
@@ -11,3 +11,9 @@ challenge-wizard-duel = Wizard duel! Fight!
 challenge-one-shot-gun = One gun and one bullet. Don't get shot.
 challenge-keep-the-crown = Punch the person with the crown to steal it. The person with the crown wins.
 challenge-acid-trip = Acid Battle! AUGUGUGUUGHHHHHH!!! IT BURNS!!! Stay alive!
+challenge-keep-the-crown = Punch the person with the crown to steal it. The person with the crown wins.
+challenge-put-that-helmet-on = PUT THAT FUCKING HELMET ON
+challenge-greytide = Greytide worldwide!
+challenge-landmine = Oops! Cargo mixed our soap and bats with landmines!
+challenge-xenos = Nanotrasen's Colonial Marines program needs new recruits to the frontlines, prove it!
+challenge-morb = It's morbing time
```
Comment (EmoGarbage404) [inline Resources/Locale/en-US/_StationWare/ware-events/challenges.ftl:19]:
```suggestion
challenge-morb = It's morbing time.
```

File: Resources/Locale/en-US/_StationWare/ware-events/challenges.ftl [hunk 6]
```diff
@@ -11,3 +11,9 @@ challenge-wizard-duel = Wizard duel! Fight!
 challenge-one-shot-gun = One gun and one bullet. Don't get shot.
 challenge-keep-the-crown = Punch the person with the crown to steal it. The person with the crown wins.
 challenge-acid-trip = Acid Battle! AUGUGUGUUGHHHHHH!!! IT BURNS!!! Stay alive!
+challenge-keep-the-crown = Punch the person with the crown to steal it. The person with the crown wins.
+challenge-put-that-helmet-on = PUT THAT FUCKING HELMET ON
```
Comment (EmoGarbage404) [inline Resources/Locale/en-US/_StationWare/ware-events/challenges.ftl:15]:
```suggestion
challenge-put-that-helmet-on = PUT THAT FUCKING HELMET ON!
```

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Pull request merged**